instrumentation/restlet/restlet-1.0/library/src/main/java/io/opentelemetry/instrumentation/restlet/v1_0/TracingFilter.java

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.instrumentation.restlet.v1_0;
+
+import static io.opentelemetry.instrumentation.restlet.v1_0.RestletSingletons.instrumenter;
+
+import io.opentelemetry.api.trace.Span;
+import io.opentelemetry.context.Context;
+import io.opentelemetry.context.Scope;
+import io.opentelemetry.instrumentation.api.tracer.ServerSpan;
+import org.restlet.Filter;
+import org.restlet.data.Request;
+import org.restlet.data.Response;
+
+public final class TracingFilter extends Filter {
+
+  private static final ThreadLocal<Scope> otelScope = new ThreadLocal<>();
+  private static final ThreadLocal<Context> otelContext = new ThreadLocal<>();
```

Comment:
Consider adding a new `ContextAndScope`/`State` class (use `@AutoValue`) and keeping it in a single thread local - you'll have one thread local lookup less for each request.

Reply:
Sure, something like `ContextAndScope` from `rocketmq` instrumentation?

Reply:
Yeah, exactly that